### PR TITLE
RSS 1.0 Update

### PR DESCRIPTION
**Pull request by nmonga91**
@jworcestBAH or @bsweger

### CODE REVIEW

File: dataactvalidator/config/awardFinancialRules.csv [hunk 1]
```diff
@@ -1,12 +1,11 @@
...
-REQUIRED_CONDITIONAL,TRUE,parentawardid,C13,,Required if C13 passes,1,
-REQUIRED_CONDITIONAL,TRUE,ParentAwardAwardingAgencyCodeNIST,C13,,Required if C13 passes,1,
-EQUAL,TRUE,awardidfieldtype,FA,,Checks if award id is FA,2,C14
-REQUIRED_CONDITIONAL,TRUE,FainAwardNumber,C14,,Required if C14 passes,1,
-REQUIRED_CONDITIONAL,TRUE,URI,C14,,Required if C14 passes,1,
+rule_type,is_single_field,field_name,rule_text_one,rule_text_two,description,rule_timing,rule_label
+car_match,FALSE,,"allocationtransferagencyidentifier, agencyidentifier, beginningperiodofavailability, endingperiodofavailability, availabilitytypecode, mainaccountcode, subaccountcode","allocation_transfer_agency, agency_identifier, beginning_period_of_availability, ending_period_of_availability, availability_type_code, main_account_code, sub_account_code",TAS fields must match CAR file,1,A1
+MIN LENGTH,TRUE,ParentAwardId,1,,Checks if parent award ID is empty,2,C10
+REQUIRED_CONDITIONAL,TRUE,ParentAwardAwardingAgencyCode(FPDS),C10,,Required if C10 passes,1,
+IN_SET,TRUE,awardidfieldtype,"P3,P7",,Checks if award id is either P3 or P7,2,C13
```
Comment (bsweger) [inline dataactvalidator/config/awardFinancialRules.csv]: Should the description for this rule read "Checks if award id field type is P3 or P7"?

File: dataactvalidator/config/awardFinancialRules.csv [hunk 2]
```diff
@@ -1,12 +1,11 @@
...
+REQUIRED_CONDITIONAL,TRUE,piidfpds,C13,,Required if C13 passes,1,
+REQUIRED_CONDITIONAL,TRUE,parentawardid,C13,,Required if C13 passes,1,
+REQUIRED_CONDITIONAL,TRUE,ParentAwardAwardingAgencyCodeFPDS,C13,,Required if C13 passes,1,
+EQUAL,TRUE,awardidfieldtype,FA,,Checks if award id is FA,2,C14
```
Comment (bsweger) [inline dataactvalidator/config/awardFinancialRules.csv]: @nmonga91 can you double check this EQUAL rule for awardidfieldtype? I don't see a validation rule specified for this field in the RSS 1.0 spreadsheet.

Comment (bsweger) [inline dataactvalidator/config/awardFinancialRules.csv]: Perhaps I don't fully understand how the rules are working. I see below that there are some rules that get involved if awardidfieldtype = 'FA'. Does that mean that awardidfieldtype *has* to be FA to pass, or is that check this there are a trigger for the rules below?

Comment (nmonga91) [inline dataactvalidator/config/awardFinancialRules.csv]: those are part of conditionally required. That rule was already present prior to this update so I didn't mess with it.

Comment (bsweger) [inline dataactvalidator/config/awardFinancialRules.csv]: I get that about not messing with the existing rules. Will check in w/ James and report back.

File: tests/awardFinancialMixed.csv [hunk 3]
```diff
@@ -1,17 +1,16 @@
...
+,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
```
Comment (bsweger) [inline tests/awardFinancialMixed.csv]: Double checking: Is the lack of line break after the header column something in there specifically to test some of the new header error logic?

File: tests/awardFinancialValid.csv [hunk 4]
```diff
@@ -1,11 +1,11 @@
...
...unpaidtotal_cpe,ussgl490100deliveredordersobligationsunpaid_fyb,ussgl490100deliveredordersobligationsunpaid_cpe,ussgl493100deliveredordersobligationstransferredunpaid_cpe,ussgl498100upward_adjustmentsofprioryeardeliveredordersobligationsunpaid_cpe,outlayamountbyaward_fyb,outlayamountbyaward_cpe,outlaysundeliveredordersprepaidtotal_fyb,outlaysundeliveredordersprepaidtotal_cpe,ussgl480200undeliveredordersobligationsprepaidadvanced_fyb,ussgl480200undeliveredordersobligationsprepaidadvanced_cpe,ussgl483200undeliveredordersobligationstransferredprepaidadvanced_cpe,ussgl488200upwardadjustmentsofprioryearundeliveredordersobligationsprepaidadvanced_cpe,outlaysdeliveredorderspaidtotal_fyb,outlaysdeliveredorderspaidtotal_cpe,ussgl490200deliveredordersobligationspaid_cpe,ussgl490800authorityoutlayednotyetdisbursed_fyb,ussgl490800authorityoutlayednotyetdisbursed_cpe,ussgl498200upwardadjustmentsofprioryeardeliveredordersobligationspaid_cpe19,72,0,0,X,306,0,P7,ABCD,ABCD,ABCD,ABCD,123456798,12-34-56-78-90-AS-DF-AB-XZ-YW,School improvement grants,2,25.4,D,6500,6500,6500,6500,6500,6500,6500,6500,6500,6500,6500,6500,6500,6500,6500,6500,6500,6500,6500,6500,6500,6500,6500,6500,6500,6500,6500,6500,6500
```
Comment (bsweger) [inline tests/awardFinancialValid.csv]: Hmmm, same issue here re: missing a line break before the first row of test data. Tests are running, but probably b/c the validator ignores extraneous columns in the header. I suspect the first row of test data isn't running through the validator.

File: dataactvalidator/config/awardFinancialRules.csv (hunk 2 above)
Comment (bsweger) [inline dataactvalidator/config/awardFinancialRules.csv]: @nmonga91 James confirmed that there's a timing element to this validation rule that should ensure things are working as expected, so we should be good on this.

**Comment by bsweger** [comment]
@nmonga91 Finished reviewing everything but fileTypeTests and baseTests. I'll look at those once the dev branch merge is worked out.

**Comment by nmonga91** [comment]
@bsweger Fixed fileTypeTests and baseTests

File: dataactvalidator/config/awardFinancialRules.csv (hunk 1 above)
Comment (jworcestBAH) [inline dataactvalidator/config/awardFinancialRules.csv]: Yes, it should be updated to say that, this was my mistake in an earlier version

Comment (jworcestBAH) [inline dataactvalidator/config/awardFinancialRules.csv]: Pushed the fix to this.

File: tests/awardFinancialValid.csv (hunk 4 above)
Comment (jworcestBAH) [inline tests/awardFinancialValid.csv]: The current code will ignore all the extra stuff after the normal headers, but we should probably remove that, it was not intentionally placed there.

**Comment by bsweger** [comment]
Ok, everything looks good now.  Checked:
* the raw .csv files against Treasury's RSS 1.0 spec
* our resulting config files against the raw .csvs
* the rules we've created so far against the RSS 1.0 spec

**Pull request merged**